src/kit/RadioGroup.tsx

```diff
@@ -0,0 +1,136 @@
+import { Radio } from 'antd';
+import { RadioChangeEvent } from 'antd/lib/radio';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import Icon, { IconName, IconSize } from 'kit/Icon';
+import useResize from 'kit/internal/useResize';
+import Tooltip from 'kit/Tooltip';
+
+import { ConditionalWrapper } from './internal/ConditionalWrapper';
+import css from './RadioGroup.module.scss';
+
+export const DEFAULT_RESIZE_THROTTLE_TIME = 500;
+
+interface Props {
+  defaultValue?: string;
+  iconOnly?: boolean;
+  onChange?: (id: string) => void;
+  options: RadioGroupOption[];
+  value?: string;
+  radioType?: 'button' | 'radio';
+}
+
+export interface RadioGroupOption {
+  icon?: IconName;
+  iconSize?: IconSize;
+  id: string;
+  label: string;
+}
+
+interface SizeInfo {
+  baseHeight: number;
+  baseWidth: number;
+  parentWidth: number;
+}
+
+const PARENT_WIDTH_BUFFER = 16;
+const HEIGHT_LIMIT = 50;
+
+const RadioGroup: React.FC<Props> = ({
+  defaultValue,
+  iconOnly = false,
+  onChange,
+  options,
+  value,
+  radioType = 'button',
+}: Props) => {
+  const baseRef = useRef<HTMLDivElement>(null);
+  const originalWidth = useRef<number>();
+  const { refCallback, size } = useResize();
+  const [sizes, setSizes] = useState<SizeInfo>({ baseHeight: 0, baseWidth: 0, parentWidth: 0 });
+  const classes = [css.base];
+
+  const hasIconsAndLabels = useMemo(() => {
+    if (options.length === 0) return false;
+    return options.reduce((acc, option) => acc || (!!option.icon && !!option.label), false);
+  }, [options]);
+
+  const showLabels = useMemo(() => {
+    if (!hasIconsAndLabels || !baseRef.current) return true;
+    if (sizes.baseWidth === 0 || sizes.parentWidth === 0) return true;
+    if (originalWidth.current && originalWidth.current < sizes.parentWidth) return true;
+    if (sizes.baseHeight > HEIGHT_LIMIT) return false;
+
+    return true;
+  }, [hasIconsAndLabels, sizes]);
+  if (iconOnly) classes.push(css.iconOnly);
+
+  const handleChange = useCallback(
+    (e: RadioChangeEvent) => {
+      if (onChange) onChange(e.target.value);
+    },
+    [onChange],
+  );
+
+  useEffect(() => refCallback(baseRef.current), [refCallback]);
```

Comment:
doesn't `useResize` return a normal ref as well? I recommended wrapping the ref callback so you could get the parent reference at the same time instead of waiting for the render to finish.